srai/loaders/osm_tag_loader/osm_tag_loader.py

```diff
@@ -0,0 +1,113 @@
+"""
+OSM Tag Loader.
+
+This module contains loader capable of loading OpenStreetMap tags.
+
+"""
+from itertools import product
+from typing import Dict, List, Tuple, Union
+
+import geopandas as gpd
+import osmnx as ox
+import pandas as pd
+from functional import seq
+from tqdm import tqdm
+
+from srai.utils.constants import WGS84_CRS
+
+
+class OSMTagLoader:
+    """
+    OSMTagLoader.
+
+    OSM(OpenStreetMap)[1] tag loader is a loader capable of downloading objects
+    from a given area from OSM. Tags in this context mean arbitrary
+    key:value pairs, that are used by OSM users to give meaning to geometries.
+
+    This loader is a wrapper around the `osmnx` library. It uses `osmnx.geometries_from_polygon`
+    to make individual queries.
+
+    References:
+        1. https://www.openstreetmap.org/
+
+    """
+
+    _PBAR_FORMAT = "Downloading {}: {}"
+
+    def load(
+        self,
+        area: gpd.GeoDataFrame,
+        tags: Dict[str, Union[List[str], str, bool]],
+    ) -> gpd.GeoDataFrame:
+        """
+        Download OSM objects with specified tags for a given area.
+
+        The loader first downloads all objects with `tags`. It returns a GeoDataFrame containing
+        the `geometry` column and columns for tag keys.
+
+        Note: some key/value pairs might be missing from the resulting GeoDataFrame,
+            simply because there are no such objects in the given area.
+
+        Args:
+            area (gpd.GeoDataFrame): Area for which to download objects.
+            tags (Dict[str, Union[List[str], str, bool]]): A dictionary
+                specifying which tags to download.
+                The keys should be OSM tags (e.g. `building`, `amenity`).
+                The values should either be `True` for retrieving all objects with the tag,
+                string for retrieving a single tag-value pair
+                or list of strings for retrieving all values specified in the list.
+                `tags={'leisure': 'park}` would return parks from the area.
+                `tags={'leisure': 'park, 'amenity': True, 'shop': ['bakery', 'bicycle']}`
+                would return parks, all amenity types, bakeries and bicycle shops.
+
+        Returns:
+            gpd.GeoDataFrame: Downloaded objects as a GeoDataFrame.
+
+        """
+        area_wgs84 = area.to_crs(crs=WGS84_CRS)
+
+        _tags = self._flatten_tags(tags)
+
+        total_tags_num = len(_tags)
+        total_queries = len(area) * total_tags_num
+
+        max_key_value_name_len = self._get_max_key_value_name_len(_tags)
+        max_desc_len = max_key_value_name_len + len(self._PBAR_FORMAT.format("", ""))
```

Comment:
I think rewording this may make it easier to read
```suggestion
        key_value_name_max_len = self._get_key_value_name_max_len(_tags)
        desc_max_len = key_value_name_max_len + len(self._PBAR_FORMAT.format("", ""))
```